solver/llbsolver/vertex.go

```diff
@@ -79,17 +79,21 @@ func NormalizeRuntimePlatforms() LoadOpt {
 					OS:           p.OS,
 					Architecture: p.Architecture,
 					Variant:      p.Variant,
+					OSVersion:    p.OSVersion,
+					OSFeatures:   p.OSFeatures,
 				}
 			}
 			op.Platform = defaultPlatform
 		}
-		platform := ocispecs.Platform{OS: op.Platform.OS, Architecture: op.Platform.Architecture, Variant: op.Platform.Variant}
+		platform := ocispecs.Platform{OS: op.Platform.OS, Architecture: op.Platform.Architecture, Variant: op.Platform.Variant, OSVersion: op.Platform.OSVersion, OSFeatures: op.Platform.OSFeatures}
 		normalizedPlatform := platforms.Normalize(platform)
 
 		op.Platform = &pb.Platform{
 			OS:           normalizedPlatform.OS,
 			Architecture: normalizedPlatform.Architecture,
 			Variant:      normalizedPlatform.Variant,
+			OSVersion:    normalizedPlatform.OSVersion,
+			OSFeatures:   normalizedPlatform.OSFeatures,
 		}
 
 		return nil
```

Comment:
For the record, my testing with `FROM` a manifest list (the known failure case) only required this change to pass. However, I decided to cover every `OCI.Platform` instance I could since a bunch of them are just copying structures around, and I also thought it might be handy for a Dockerfile to be able to fork not just on architecture and OS, but on OS Version, since that tends to be important in Windows.